File: BOG/python/2884.py
time, minute = map(int, input().split(" "))

if time > 0 and minute >= 45:
    print(str(time) + ' ' + str(minute - 45))
elif time > 0 and minute < 45:
    print(str(time - 1) + ' ' + str(minute + 15))
elif time <= 0 and minute >= 45 :
    print(str(time) + ' ' + str(minute - 45))
elif time <= 0 and minute < 45:
    print(str(time + 23) + ' ' + str(minute + 15))

# print(str(time) + ' ' + str(minute))

File: BOG/javascript/2884.js
// testCase = require('fs').readFileSync('./dev/stdin').toString().trim()
testCase = require('fs').readFileSync('예제.txt').toString().trim()

const [hour, minute] = testCase.split(' ').map(Number)

if(minute<45 && hour>0){
    hour-1
    minute+15
    console.log(hour + ' ' + minute)
}
else{
    hour+23
    minute-45
    console.log(hour + ' ' + minute)
}
